sw/device/silicon_creator/lib/drivers/otbn.c

```diff
@@ -59,15 +59,32 @@ static rom_error_t check_offset_len(uint32_t offset_bytes, size_t num_words,
   return kErrorOk;
 }
 
+bool otbn_is_busy(void) {
+  uint32_t status = abs_mmio_read32(kBase + OTBN_STATUS_REG_OFFSET);
+  return status != kOtbnStatusIdle && status != kOtbnStatusLocked;
+}
+
 /**
  * Helper function for writing to OTBN's DMEM or IMEM.
  *
  * @param dest_addr Destination address.
  * @param src Source buffer.
  * @param num_words Number of words to copy.
  */
-static void otbn_write(uint32_t dest_addr, const uint32_t *src,
-                       size_t num_words) {
+static rom_error_t otbn_write(uint32_t dest_addr, const uint32_t *src,
+                              size_t num_words) {
+  // If OTBN is busy, wait for it to be done.
```

Comment:
nit: I think it's enough to perform this check once before starting to use OTBN, e.g. in `mod_exp_otbn.c`